docs/5.Minting-Module.md

```diff
@@ -0,0 +1,693 @@
+}
+```
+
+**expected_keepers.go** Here we import the Rebalancer module to get access to the control parameter.
+
+```golang
+package types // noalias
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/supply/exported"
+)
+
+// StakingKeeper defines the expected staking keeper
+type StakingKeeper interface {
+	StakingTokenSupply(ctx sdk.Context) sdk.Int
+	BondedRatio(ctx sdk.Context) sdk.Dec
+}
+
+// SupplyKeeper defines the expected supply keeper
+type SupplyKeeper interface {
+	GetModuleAddress(name string) sdk.AccAddress
+
+	// TODO remove with genesis 2-phases refactor https://github.com/cosmos/cosmos-sdk/issues/2862
+	SetModuleAccount(sdk.Context, exported.ModuleAccountI)
+
+	SendCoinsFromModuleToAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error
+	SendCoinsFromModuleToModule(ctx sdk.Context, senderModule, recipientModule string, amt sdk.Coins) error
+	MintCoins(ctx sdk.Context, name string, amt sdk.Coins) error
+}
+
+type RebalancerKeeper interface {
+    GetValidatorControlParameter() sdk.Dec
+}
+```
+
+**params.go** Here we add the ValidatorControlParam to the mint module's parameters.
+
+```golang
+package types
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// Parameter store keys
+var (
+	KeyMintDenom           		 = []byte("MintDenom")
+	KeyInflationRateChange       = []byte("InflationRateChange")
+	KeyInflationMax              = []byte("InflationMax")
+	KeyInflationMin              = []byte("InflationMin")
+	KeyGoalBonded                = []byte("GoalBonded")
+	KeyBlocksPerYear             = []byte("BlocksPerYear")
+	KeyValidatorControlParam     = []byte("ValidatorControlParam")
+)
+
+// ParamTable for minting module.
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func NewParams(
+	mintDenom string, inflationRateChange, inflationMax, inflationMin, goalBonded sdk.Dec, blocksPerYear uint64, validatorControlParam sdk.Dec
+) Params {
+
+	return Params{
+		MintDenom:                    mintDenom,
+		InflationRateChange:          inflationRateChange,
+		InflationMax:                 inflationMax,
+		InflationMin:                 inflationMin,
+		GoalBonded:                   goalBonded,
+		BlocksPerYear:                blocksPerYear,
+		ValidatorControlParam:        validatorControlParam
+	}
+}
+
+// default minting module parameters
+func DefaultParams() Params {
+	return Params{
+		MintDenom:                   sdk.DefaultBondDenom,
+		InflationRateChange:         sdk.NewDecWithPrec(13, 2),
+		InflationMax:                sdk.NewDecWithPrec(20, 2),
+		InflationMin:                sdk.NewDecWithPrec(7, 2),
+		GoalBonded:                  sdk.NewDecWithPrec(67, 2),
+		BlocksPerYear:               uint64(60 * 60 * 8766 / 5), // assuming 5 second block times
```

Comment:
This might change , if the number of trasactions increase . Would that affect the calculation since we are assuming this to be a constant  .

Reply:
This is pulled straight from the cosmos mint module. 

I'll add to the doc to make clearer what the actual changers are vs what is standard cosmos